File: Compiler/TranslatorTests/TestProjects/18/TestIssue434.cs
using Bridge;

namespace TestIssue434
{
    public class Issue434A
    {
        [Init]
        public static void Method1()
        {
            DoSomething(1);
        }

        [Init(InitPosition.Before)]
        public static void Method2()
        {
            DoSomething(2);
        }

        [Init(InitPosition.After)]
        public static void Method3()
        {
            DoSomething(3);
        }

        [Init()]
        public static void Method4()
        {
            DoSomething(4);
        }

        private static void DoSomething(int i)
        {
            Bridge.Html5.Console.Log(i);
        }
    }

    public class Issue434B
    {
        [Init]
        public static void Method1()
        {
            DoSomething(1);
        }

        [Init(InitPosition.Before)]
        public static void Method2()
        {
            DoSomething(2);
        }

        [Init(InitPosition.After)]
        public static void Method3()
        {
            DoSomething(3);
        }

        [Init()]
        public static void Method4()
        {
            DoSomething(4);
        }

        private static void DoSomething(int i)
        {
            Bridge.Html5.Console.Log(i);
        }
    }
}

File: Compiler/TranslatorTests/TestProjects/18/Bridge/reference/testIssue434.js
// Top

    Bridge.init(function(){
        TestIssue434.Issue434A.doSomething(2);
    });

    Bridge.define("TestIssue434.Issue434A", {
        statics: {
            method1: function () {
                TestIssue434.Issue434A.doSomething(1);
            },
            method3: function () {
                TestIssue434.Issue434A.doSomething(3);
            },
            method4: function () {
                TestIssue434.Issue434A.doSomething(4);
            },
            doSomething: function (i) {
                Bridge.Console.log(i);
            }
        }
    });

    Bridge.init(function() { TestIssue434.Issue434A.method1(); });
    Bridge.init(function() { TestIssue434.Issue434A.method3(); });
    Bridge.init(function() { TestIssue434.Issue434A.method4(); });

    Bridge.init(function(){
        TestIssue434.Issue434B.doSomething(2);
    });

    Bridge.define("TestIssue434.Issue434B", {
        statics: {
            method1: function () {
                TestIssue434.Issue434B.doSomething(1);
            },
            method3: function () {
                TestIssue434.Issue434B.doSomething(3);
            },
            method4: function () {
                TestIssue434.Issue434B.doSomething(4);
            },
            doSomething: function (i) {
                Bridge.Console.log(i);
            }
        }
    });

    Bridge.init(function() { TestIssue434.Issue434B.method1(); });
    Bridge.init(function() { TestIssue434.Issue434B.method3(); });
    Bridge.init(function() { TestIssue434.Issue434B.method4(); });

    Bridge.define("TestIssue434.Issue434C");

// Bottom
